src/analysis/inner_product_space/adjoint.lean

```diff
@@ -186,6 +186,43 @@ lemma is_adjoint_pair (A : E' →L[ℝ] F') :
 
 end real
 
+section unitary
+
+variables {U : (E →L[𝕜] E)}
+
+/-- Builds a linear equivalence from `U ∈ unitary (E →L[𝕜] E)`. -/
+def linear_equiv_of_unitary (hU : U ∈ unitary (E →L[𝕜] E)) : (E ≃ₗ[𝕜] E) :=
+  { to_fun    := U,
+    map_add'  := map_add U,
+    map_smul' := map_smul U,
+    inv_fun   := (star U : (E →L[𝕜] E)),
+    left_inv  := by { rw function.left_inverse_iff_comp, ext, simp only [id.def, function.comp_app,
+      ← continuous_linear_map.mul_apply, unitary.star_mul_self_of_mem hU,
+      continuous_linear_map.one_apply] },
+    right_inv := by { rw function.right_inverse_iff_comp, ext, simp only [id.def, function.comp_app,
+      ← continuous_linear_map.mul_apply, unitary.mul_star_self_of_mem hU,
+      continuous_linear_map.one_apply] } }
```

Comment:
Please unindent this block

Reply:
Got it.  I think I understand the convention:  this is similar to not indenting begin/end blocks.